### PR TITLE
Option initialization fix + Avoids empty string app cap.

### PR DESCRIPTION
**Pull request by sebv**
ping @jlipps @moizjv @Jonahss @imurchie @scottdixon , gettng rid of `app === ''` case, it's better to get a Null/Undefined exception and fix it.

### CODE REVIEW

**Comment by sebv** [comment]
Also fixes initialOpts bug.

File: lib/commands/session.js [hunk 1]
```diff
@@ -13,13 +13,17 @@ commands.createSession = async function (caps) {
...
-  // keep track of initial opts since we're going to mutate opts
-  this.initialOpts = _.clone(this.opts);
+  this.opts = _.cloneDeep(this.initialOpts);
 
   // merge caps onto opts so we don't need to worry about what's where
   Object.assign(this.opts, this.caps);
 
+
+  // Prevents empty string caps so we don't need to test it everywhere
+  if(typeof this.opts.app === 'string' && this.opts.app.trim() === '') {
```
Comment (jlipps) [inline lib/commands/session.js]: `if (`

File: lib/helpers.js [hunk 2]
```diff
@@ -12,9 +12,11 @@ import request from 'request';
 const ZIP_EXTS = ['.zip', '.ipa'];
 
 async function configureApp (app, appExt) {
+  if(_.isNull(app) || _.isUndefined(app)) { return; }
+
   let newApp = null;
   let shouldUnzipApp = _.contains(ZIP_EXTS, path.extname(app));
-  if (app.substring(0, 4).toLowerCase() === "http") {
+  if (typeof app === 'string' && app.substring(0, 4).toLowerCase() === "http") {
```
Comment (jlipps) [inline lib/helpers.js]: could do this, or:

```js
if ((app || '').substring(0, 4).toLowerCase() === "http") {
```

**Comment by sebv** [comment]
fixed.

**Comment by jlipps** [comment]
LGTM

**Pull request merged**